### PR TITLE
implement numeric max penalty setting. Closes #603

### LINKED ISSUE #603: Allow disabling colored shapes
Currently if colors and shapes are both enabled some of the shape hats will also have colors. This allows additional tokens to have hats but results in longer selectors (i.e. "take blue bolt bat"). Additionally it is harder to determine the color of some shapes than others. For example the bolt shape has far less area than the default circle. Therefore I suggest adding a setting to disable colored shapes so that when colors and shapes are both enabled only one or the other will be used at time.

### PR DESCRIPTION
**Pull request by AndrewDant**
Added a new maxPenalty setting for controlling what combinations of colors and hats are allowed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -334,6 +334,11 @@
 					},
 					"additionalProperties": false
 				},
+				"cursorless.maxPenalty": {
```
Comment (pokey) [inline package.json:337]:
```suggestion
				"cursorless.maximumHatStylePenalty": {
```

File: package.json [hunk 2]
```diff
@@ -334,6 +334,11 @@
...
+					"description": "The maximum allowed total penalty, equal to the shape penalty plus the colour penalty. Values less than or equal to zero are treated as no maximum.",
```
Comment (pokey) [inline package.json:338]:
```suggestion
					"description": "The maximum allowed penalty for a hat style. Any hat style whose penalty is greater than this amount will not be used. A hat style penalty is defined to be the shape penalty plus the colour penalty. Setting this value less than or equal to zero is treated as no maximum.",
```

File: src/core/Decorations.ts [hunk 3]
```diff
@@ -185,6 +185,8 @@ export default class Decorations {
     const colorPenalties = vscode.workspace
       .getConfiguration("cursorless.hatPenalties")
       .get<Record<HatColor, number>>("colors")!;
+      const maxPenalty = vscode.workspace
+    .getConfiguration("cursorless").get<number>("maxPenalty")!;
```
Comment (pokey) [inline src/core/Decorations.ts:189]:
```suggestion
    .getConfiguration("cursorless").get<number>("maximumHatStylePenalty")!;
```

**Review by pokey — COMMENTED** [review]
Looks fine; left a couple minor comments. Also would be good to run your changes through prettier and Eslint. Those extensions should be suggested when you open cursorless workspace, but lmk if they weren't

**Review by pokey — COMMENTED** [review]
looks good!  oh yeah don't forget to update [the docs](https://github.com/cursorless-dev/cursorless#extension-settings)

**Comment by AndrewDant** [comment]
Good point, just added that.

**Comment by AndrewDant** [comment]
Closes #603

**Review by pokey — APPROVED** [review]
Looks good!

**Pull request merged**